internal/parse/parser.go

```diff
@@ -250,51 +243,67 @@ func (p *Parser) parseIdentifier() (string, bool) {
 		return "*", true
 	}
 
-	idStart := p.pos
+// parseIdentifier parses either a name made up only of nameBytes (or an
+// asterisk if asteriskF=allowAsterisk)
+func (p *Parser) parseIdentifier(asteriskF asteriskFlag) (string, bool) {
+	if asteriskF == allowAsterisk {
+		if p.skipByte('*') {
+			return "*", true
+		}
+	}
+	mark := p.pos
 	if p.skipName() {
-		return p.input[idStart:p.pos], true
+		return p.input[mark:p.pos], true
 	}
 	return "", false
 }
 
-// parseFullName parses a column name or Go field name, optionally dot-prefixed by
-// its table or type name respectively.
-func (p *Parser) parseFullName(idc idClass) (FullName, bool) {
+// parseGoObject parses a source or target go object of the form Prefix.Name
+// where the type is the Prefix and the field is the Name (this applies to maps
+// and structs).
+func (p *Parser) parseGoObject() (FullName, bool, error) {
 	cp := p.save()
 	var fn FullName
-	if id, ok := p.parseIdentifier(); ok {
-		fn.Prefix = id
+	if id, ok := p.parseIdentifier(disallowAsterisk); ok {
 		if p.skipByte('.') {
-			if id, ok := p.parseIdentifier(); ok {
-				fn.Name = id
-				return fn, true
-			}
-		} else {
-			// A column name specified without a table prefix is a name not a
-			// prefix.
-			if idc == columnId {
-				fn.Name = fn.Prefix
-				fn.Prefix = ""
+			if idField, ok := p.parseIdentifier(allowAsterisk); ok {
+				return FullName{Prefix: id, Name: idField}, true, nil
```

Comment:
We can use the short form here: `return FullName{id, idField}, true, nil`

Reply:
Changed.